doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+This document covers high level design and interaction aspects of the SONiC software components for L2 & L3 EVPN support, including Orchestration agent submodules (FdbOrch, VXLANOrch, VrfOrch, RouteOrch, etc.), SAI objects, SwSS managers (VXLAN, VRF managers), interactions with Linux kernel, FRR (BGP, Zebra), and syncd modules (fpmsyncd, fdbsyncd, neighsyncd). 
+
+
+The following aspects are outside the scope of this document.
+
+- Support for L2 multi-tenancy is not in scope. VLAN ids received on all the access ports are mapped to the same broadcast domain and hence the same VNI. 
+- IPv6 addresses for VTEPs.
+- Static VXLAN tunnels.
+- Multi-homing support described in the EVPN RFCs.
+
+
+
+
+# 2 Feature Requirements
+
+## 2.1 Functional Requirements
+
+Following requirements are addressed by the design presented in this document:
+
+1. Support creation and deletion of BGP EVPN discovered VXLAN tunnels (IPv4 only) based on IMET, MAC and IP Prefix  routes.
+2. Support origination and withdrawal of IMET routes. 
+3. Support origination and withdrawal of dynamic & static MAC routes.
+4. Support origination and withdrawal of MACIP routes (ARP, ND) 
+5. Support IPv4 & IPv6 Prefix route origination and remote route handling in user VRFs.
+6. Support Ingress Replication for L2 BUM traffic over VXLAN tunnels. Underlay IP Multicast is not supported.  
+7. Support remote MAC programming over VXLAN tunnels.
+8. Support MAC/MACIP move from local to remote, remote to local, remote to remote. 
+9. Support Routing In and Out of VXLAN Tunnel (RIOT) (MP-BGP EVPN Based)
+    - Support Asymmetric Integrated Routing and Bridging (Asymmetric IRB)
+    - Support Symmetric Integrated Routing and Bridging (Symmetric IRB)
+10. Support ARP/ND suppression.
+11. Support Tunnel ECMP and underlay path failovers.
+12. Support a common VLAN-VNI map for all the EVPN tunnels.
+
+
+
+Following requirements will be attempted on a best effort basis. 
+
+1. Support monitoring and display of tunnel operational status.
+2. Support tunnel packet/octet  tx/rx counters.
+
+
+
+## 2.2 Configuration and Management Requirements
+
+Configuration and Management for this feature will be based the proposed management framework for SONiC.
+
+1. Support configuration of Source IP address of the VTEP. 
+2. Support configuration of a global VLAN-VNI map.
+3. Support configuration of L3VNI association with VRF. 
+
+
+
+FRR version 7.0.1, and later, is assumed here. FRR configuration is assumed in "split" configuration mode.
+BGP EVPN configurations in FRR are referred wherever required in this document. However, this document does not serve as reference guide for the FRR configurations.
+
+
+
+## 2.3 Scalability Requirements
```

Comment:
These are part of an overall requirements section. Scale is controlled by hardware as well as 
FRR capabilities for the control plane.

Reply:
I'm not sure how relevant those scale requirements are because I don't understand how they affect the design. Can you please provide the justification for each of them where the number was derived from and what implication it has on a design if any?

What about CRM? Any work expected to be done in this area?

Reply:
On the Scaling - Agreed that different platforms can support different numbers.  If the suggestion is to remove the numbers and mention that this will be specified in the platform specific documentation then this can be incorporated. 

However I do see scalability numbers put in as part of the 
https://github.com/Azure/SONiC/blob/master/doc/vxlan/Vxlan_hld.md
 
On CRM - no work is planned. The CRM depends on SAI Switch attributes to be defined for VXLAN which is not defined presently.